Source/Scene/MetadataType.js

```diff
@@ -0,0 +1,285 @@
+      return 255;
+    case MetadataType.INT16:
+      return 32767;
+    case MetadataType.UINT16:
+      return 65535;
+    case MetadataType.INT32:
+      return 2147483647;
+    case MetadataType.UINT32:
+      return 4294967295;
+    case MetadataType.INT64:
+      if (FeatureDetection.supportsBigInt()) {
+        // Need to initialize with a string otherwise the value will be 9223372036854775808
+        return BigInt("9223372036854775807"); // eslint-disable-line
+      }
+      return Math.pow(2, 63) - 1;
+    case MetadataType.UINT64:
+      if (FeatureDetection.supportsBigInt()) {
+        // Need to initialize with a string otherwise the value will be 18446744073709551616
+        return BigInt("18446744073709551615"); // eslint-disable-line
+      }
+      return Math.pow(2, 64) - 1;
+    case MetadataType.FLOAT32:
+      return 340282346638528859811704183484516925440.0;
+    case MetadataType.FLOAT64:
+      return Number.MAX_VALUE;
+  }
+};
+
+/**
+ * Returns whether the type is a numeric type.
+ *
+ * @param {MetadataType} type The type.
+ * @returns {Boolean} Whether the type is a numeric type.
+ */
+MetadataType.isNumericType = function (type) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.string("type", type);
+  //>>includeEnd('debug');
+
+  switch (type) {
+    case MetadataType.INT8:
+    case MetadataType.UINT8:
+    case MetadataType.INT16:
+    case MetadataType.UINT16:
+    case MetadataType.INT32:
+    case MetadataType.UINT32:
+    case MetadataType.INT64:
+    case MetadataType.UINT64:
+    case MetadataType.FLOAT32:
+    case MetadataType.FLOAT64:
+      return true;
+    default:
+      return false;
+  }
+};
+
+/**
+ * Returns whether the type is an integer type.
+ *
+ * @param {MetadataType} type The type.
+ * @returns {Boolean} Whether the type is an integer type.
+ */
+MetadataType.isIntegerType = function (type) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.string("type", type);
+  //>>includeEnd('debug');
+
+  switch (type) {
+    case MetadataType.INT8:
+    case MetadataType.UINT8:
+    case MetadataType.INT16:
+    case MetadataType.UINT16:
+    case MetadataType.INT32:
+    case MetadataType.UINT32:
+    case MetadataType.INT64:
+    case MetadataType.UINT64:
+      return true;
+    default:
+      return false;
+  }
+};
+
+/**
+ * Returns whether the type is an unsigned integer type.
+ *
+ * @param {MetadataType} type The type.
+ * @returns {Boolean} Whether the type is an unsigned integer type.
+ */
+MetadataType.isUnsignedIntegerType = function (type) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.string("type", type);
+  //>>includeEnd('debug');
+
+  switch (type) {
+    case MetadataType.UINT8:
+    case MetadataType.UINT16:
+    case MetadataType.UINT32:
+    case MetadataType.UINT64:
+      return true;
+    default:
+      return false;
+  }
+};
+
+/**
+ * Normalizes signed integers to the range [-1.0, 1.0] and unsigned integers to
```

Comment:
Add a mention that for Big Ints, they get converted to `Number` because they do not support math operations.